### PR TITLE
[SOL-36] [FE] Create markup for viewing only the user's campaigns

### PR DESCRIPTION
**Pull request by kenorenzsun**
## Issue Link
- Backlog link
https://framgiaph.backlog.com/view/SPH_SOLIDITY-36
## Target Page | Component | Function
- /profile
- Components : user-card , tabs

## Pre-condition
- npm run nuxt:dev

## Steps | Commands
1. None

## Expected Output
- Add here

## Definition of Done
- Followed design in figma

## Test View Points
- [ ] Add here

## Notes
N/A

## Screenshots | Recordings
image

image
//github.com/framgia/sph-solidity/assets/116168014/4773947b-3872-420e-93c1-05804e17c094)

### General
- [ ] Does the code meet the requirements of the project?
- [ ] Is the code readable and maintainable?
- [ ] Are there any unnecessary or redundant lines of code?
- [ ] Are there any long methods or functions that should be refactored?
- [ ] Are there any performance issues?

### Functionality
- [ ] Are there any logical errors?
- [ ] Are edge cases handled properly?
- [ ] Are there any potential security vulnerabilities?

### Code Style
- [ ] Does the code adhere to the project's coding standards?
- [ ] Are variable and function names descriptive and meaningful?
- [ ] Is there proper use of comments and documentation?
- [ ] Is the code properly formatted? Are indentation, spacing, and formatting consistent?

### Security
- [ ] Is the website secure from common attacks?
- [ ] Are inputs properly validated and sanitized?
- [ ] Are sensitive data and credentials properly protected?

### Testing
- [ ] Are there sufficient test cases?
- [ ] Do the test cases cover all functionality?
- [ ] Are the test cases properly written and documented?

### Additional Comments
- [ ] Are there any other issues or concerns with the code?
- [ ] Are there any suggestions for improvement?
- [ ] Are there any positive aspects of the code that should be highlighted?

### CODE REVIEW

**Review by wylenjoan — DISMISSED** [review]
Approved :white_check_mark:
Review time: 15 mins
Comment: 0
Iteration: 0

Working:
- Followed Figma design

**Review by wylenjoan — DISMISSED** [review]

File: fe/src/components/base/tabs.vue [hunk 1]
```diff
@@ -0,0 +1,38 @@
+<template>
+  <div class="flex overflow-y-auto space-x-4">
+    <div
+      v-for="tab in items"
+      :key="tab"
+      :class="`rounded-md h-11 ${
+        tab === activeTab
+          ? 'bg-gradient-to-r from-[#CF7A1E] to-[#AC3971] p-[1px] cursor-not-allowed'
```
Comment (jasonchuasun) [inline fe/src/components/base/tabs.vue:8]: Notice that this style gradient style is repeating. 
This is where we can utilize the tailwind.css custom class. 
Transfer this gradient style to tailwind.css file and create a custom class for it.

Comment (jasonchuasun) [inline fe/src/components/base/tabs.vue]:
```suggestion     
```
Notice that this style gradient style is repeating. 
This is where we can utilize the tailwind.css custom class. 
Transfer this gradient style to tailwind.css file and create a custom class for it.

**Comment by ipilikent** [comment]
## Approved PR
-  All displayed the same as the Figma design
- Can click the tabs
- The profile card is displayed
- The pagination is clickable and displayed
- The campaigns card is displayed
- The footer is displayed

## Proof
<details>

Sol- 19 - FE Viewing user campaigns

</details>

Test Case: https://docs.google.com/spreadsheets/d/114GrEn-LQQ8saIvjz9zqenuk05qmO4dN/edit#gid=1122662191
-----
- Total Test time: 0.5 hr.
- Total Comments: 0
- Total Iterations: 0

**Review by ipilikent — APPROVED** [review]

**Review by ipilikent — APPROVED** [review]

**Review by wylenjoan — DISMISSED** [review]

**Review by ipilikent — APPROVED** [review]

**Review by ipilikent — APPROVED** [review]

**Review by wylenjoan — APPROVED** [review]

**Review by johnpaul-sun — APPROVED** [review]

**Review by ipilikent — APPROVED** [review]

File: fe/src/pages/profile.vue [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+import UserCardSample from "~/mocks/user-card-sample.json";
+import CampaignCardProps from "~/types/CampaignCardProps";
+import CardSample from "~/mocks/card-sample.json";
+
+const userCardValueSample = ref<UserCardProps>(UserCardSample);
+
+const { userId, imgSrc, blockChainLabel, blockChainValue } =
+  userCardValueSample.value;
+
+const first = ref(5);
```
Comment (jasonchuasun) [inline fe/src/pages/profile.vue:55]: Name is too vague and not clear as to what this is.
Please update the name of this constant.

File: fe/src/pages/profile.vue [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+        :items="profileTabs"
+        @change-tab="activeTab = $event"
+      />
+    </div>
+    <div v-if="activeTab === 'My Campaigns'">
+      <div
+        class="mt-6 grid grid-cols-3 gap-6 sm:grid-cols-1 lg:grid-cols-2 xl:grid-cols-3 campaigns"
+      >
+        <base-campaign-card
+          v-for="i in 6"
```
Comment (jasonchuasun) [inline fe/src/pages/profile.vue:23]: Improvement:
Noticed the use of magic numbers such as this. 
```suggestion
          const numberOfCards = 6
          v-for="i in numberOfCards"
```
For future tasks, we can create a sample json file in mocks folder for instances like this. 
To simulate actual data and for easier integration in the future.

File: fe/src/pages/profile.vue (hunk 2 above)
Comment (jasonchuasun) [inline fe/src/pages/profile.vue:55]: Name is too vague and not clear as to what this is.
Please update the name of this constant to something more descriptive.
Example only: 

```suggestion
const firstPage = ref(5);
```

**Review by jasonchuasun — COMMENTED** [review]
Minor comments.

**Pull request merged**